Userland/Libraries/LibJS/JIT/Compiler.cpp

```diff
@@ -404,9 +403,45 @@ static Value cxx_decrement(VM& vm, Value value)
 void Compiler::compile_decrement(Bytecode::Op::Decrement const&)
 {
     load_accumulator(ARG1);
+
+    Assembler::Label end {};
+    Assembler::Label slow_case {};
+
+    branch_if_int32(ARG1, [&] {
+        // GPR0 = ARG1 & 0xffffffff;
+        m_assembler.mov(
+            Assembler::Operand::Register(GPR0),
+            Assembler::Operand::Register(ARG1));
+        m_assembler.mov(
+            Assembler::Operand::Register(GPR1),
+            Assembler::Operand::Imm(0xffffffff));
+        m_assembler.bitwise_and(
+            Assembler::Operand::Register(GPR0),
+            Assembler::Operand::Register(GPR1));
```

Comment:
```suggestion
        m_assembler.mov32(
            Assembler::Operand::Register(GPR0),
            Assembler::Operand::Register(ARG1));
```

Reply:
This might need an `Assembler::Extension::SignExtend`. Please add a test for this edge-case to verify it works.